### DISCUSSION TITLE
What is the current status of this project working on Unifi OS 3.x?

### DISCUSSION THREAD

**Discussion by mauvehed** [SierraSoftworks/tailscale-udm#72] [Q&A] [answered]
I run a UDM Pro (non-SE) and am upgraded to the latest 3.x OS version. I'm wondering what the current status is of this project and if it works on the latest OS. 

I know during the UDM Pro migration to 2.x, it was in beta and when I tested it, not super functional. Hoping this has changed recently!

**Comment by milesd** [comment]
I'm currently running this on 26 different controllers, at least one of each UCKG2, UCK Plus, UDM Pro and UDM SE. Haven't run into any problems.

**Comment by notheotherben** [comment, marked as answer]
I'm the author here and run this on UniFi OS 3.x without any notable issues, though from time to time OS upgrades do require manual intervention to re-install Tailscale (still working on resolving this, it's a side effect of the way that the OS image is upgraded).

**Reply by ksc98** [reply]
Are you able to ping other tailnet devices, from the UDM?  If you are, how'd you get it working?  Ideally I'd like the dream machine to also act as the dns server so when client devices on the network try to resolve tailnet hosts, they can do so, without having tailscale installed.

**Reply by notheotherben** [reply]
Have a look at #51 - this scenario isn't supported natively in this package for several reasons detailed in #68, however it looks like folks have been able to get this working in some scenarios with custom `iptables` configuration.

**Comment by mauvehed** [comment]
Great to hear, guess I’ll need to test again. Does exit-node and subnet routing work as well?
